### ISSUE TITLE
Atomic move during entity write fails

### ISSUE THREAD

**Issue by carltonwhitehead** [caeos/snoozle#24]
```no-highlight
Exception in thread "main" java.nio.file.AtomicMoveNotSupportedException: /tmp/tmp7406474855444397361.tmp -> /home/{specifics not relevant}: Invalid cross-device link
	at sun.nio.fs.UnixCopyFile.move(UnixCopyFile.java:394)
	at sun.nio.fs.UnixFileSystemProvider.move(UnixFileSystemProvider.java:262)
	at java.nio.file.Files.move(Files.java:1395)
	at org.coner.snoozle.db.entity.EntityResource.write(EntityResource.kt:80)
	at org.coner.snoozle.db.entity.EntityResource.put(EntityResource.kt:47)
```

**Comment by carltonwhitehead** [comment]
The resource is attempting an atomic move across filesystem mount points, of course it's going to fail. "How did this ever work?" 

Will need to adjust the strategy. The original intention was to ensure that any file watches only trigger with a fully-written file. Don't want a file watch to pick up a file mid-write, read some junk data in, and blow up unexpectedly, or even just waste any of its cycles only to discover the watch event is bunk.

Idea:
1. Use a `database.root.resolve("tmp")` folder
2. Do the first write into the database-scoped tmp folder
3. Do the second write atomically into the final location, with a try-catch fallback to non-atomic just in case.

**Comment by carltonwhitehead** [comment]
Going to keep this a minor fix in 0.3, instead of blowing up its scope into a breaking change rolled up in 0.4. Will implement 3b.

It would be helpful to have a tmp folder for other features like local transactions that could be guaranteed to commit atomically, see #26

**Comment by carltonwhitehead** [comment]
The tests didn't catch this because they were using a materially different scenario in terms of same-filesystem operations than would be typical in real use. The initial write (both real and tests) was _always_ to a file in a temp filesystem, but the final atomic-required write varied (real vs tests) whether it was to the temp filesystem or elsewhere. Test fixtures are set up in the temp filesystem for fast execution, so this was not triggering this behavior.

To fix this, the initial write is to a `.tmp`-suffixed sibling of the final write, so the move operation can be on the same filesystem. 

If the filesystem where the database is stored doesn't support atomic move operations, get a new filesystem.
